manifests/authoritative.pp

```diff
@@ -10,6 +10,21 @@
 
   include "powerdns::backends::${powerdns::backend}"
 
+  if ($powerdns::authoritative_group != undef) {
+    $authoritative_config_parameters = {
+      group => $powerdns::authoritative_group,
+    }
+  } else {
+    $authoritative_config_parameters = {}
+  }
+  file { $powerdns::authoritative_config:
+    ensure => 'file',
+    owner  => 'root',
+    mode   => '0640',
+    before => Service['pdns'],
+    *      => $authoritative_config_parameters,
+  }
```

Comment:
Why not keep this simple? If it's undef, it has no effect. If it does, then that's what you want.
```suggestion
  file { $powerdns::authoritative_config:
    ensure => 'file',
    owner  => 'root',
    group  => $powedns::authoritative_group,
    mode   => '0640',
    before => Service['pdns'],
  }
```

Reply:
I hadn't realized it would work that way - done!